app/submitter/converter.py

```diff
@@ -1,134 +1,147 @@
-                      SubmitterConstants.INSTRUMENT_KEY: metadata_store['form_type'],
-                      SubmitterConstants.PERIOD_KEY: metadata_store['period_id']}
-
-        paradata = {}
-        submitted_at = datetime.now(timezone.utc)
-
-        payload = {SubmitterConstants.TX_ID: metadata_store['tx_id'],
-                   SubmitterConstants.TYPE_KEY: SubmitterConstants.TYPE,
-                   SubmitterConstants.VERSION_KEY: SubmitterConstants.VERSION,
-                   SubmitterConstants.ORIGIN_KEY: SubmitterConstants.ORIGIN,
-                   SubmitterConstants.SURVEY_ID_KEY: survey_id,
-                   SubmitterConstants.SUBMITTED_AT_KEY: submitted_at.isoformat(),
-                   SubmitterConstants.COLLECTION_KEY: collection,
-                   SubmitterConstants.METADATA_KEY: metadata,
-                   SubmitterConstants.PARADATA_KEY: paradata,
-                   SubmitterConstants.DATA_KEY: data}
-
-        logging.debug("Converted answer ready for submission %s", json.dumps(payload))
-        return payload, submitted_at
-
-    @staticmethod
-    def _encode_value(value):
-        if isinstance(value, int):
-            return str(value)
-        elif isinstance(value, datetime):
-            return value.strftime(settings.SDX_DATE_FORMAT)
-        else:
-            return value
+def convert_answers(metadata, questionnaire, answer_store, routing_path):
+    """
+    Create the JSON answer format for down stream processing
+
+    :param metadata: metadata for the questionnaire
+    :param questionnaire: the questionnaire schema
+    :param answer_store: the users answers
+    :param routing_path: the path followed by the user when answering the questionnaire
+    :return: a JSON object in the following format:
+      {
+        "tx_id": "0f534ffc-9442-414c-b39f-a756b4adc6cb",
+        "type" : "uk.gov.ons.edc.eq:surveyresponse",
+        "version" : "0.0.1",
+        "origin" : "uk.gov.ons.edc.eq",
+        "survey_id": "021",
+        "collection":{
+          "exercise_sid": "hfjdskf",
+          "instrument_id": "yui789",
+          "period": "2016-02-01"
+        },
+        "submitted_at": "2016-03-07T15:28:05Z",
+        "metadata": {
+          "user_id": "789473423",
+          "ru_ref": "432423423423"
+        },
+        "data": {}
+      }
+    """
+    survey_id = questionnaire.survey_id
+    submitted_at = datetime.now(timezone.utc)
+    payload = {
+               'tx_id': metadata['tx_id'],
+               'type': 'uk.gov.ons.edc.eq:surveyresponse',
+               'version': questionnaire.data_version,
+               'origin': 'uk.gov.ons.edc.eq',
+               'survey_id': survey_id,
+               'submitted_at': submitted_at.isoformat(),
+               'collection': _build_collection(metadata),
+               'metadata': _build_metadata(metadata),
+    }
+    if questionnaire.data_version == '0.0.2':
```

Comment:
yeh, maybe the else makes it implicit rather than explicit. could potentially do something like this. http://stackoverflow.com/questions/60208/replacements-for-switch-statement-in-python